### PR TITLE
[MM-58510] Migrate tooltips of "components/file_preview_modal/file_preview_modal_main_actions/file_preview_modal_main_actions.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27236
Jira https://mattermost.atlassian.net/browse/MM-58510

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/file_preview_modal/file_preview_modal_main_actions/file_preview_modal_main_actions.tsx [hunk 1]
```diff
@@ -96,17 +91,13 @@ const FilePreviewModalMainActions: React.FC<Props> = (props: Props) => {
...
     const publicLink = (
-        <OverlayTrigger
-            delayShow={Constants.OVERLAY_TIME_DELAY}
+        <WithTooltip
+            id='link-variant-icon-tooltip.text'
             key='filePreviewPublicLink'
             placement={tooltipPlacement}
+            title={publicTooltipMessage}
             shouldUpdatePosition={true}
             onExit={() => setPublicLinkCopied(false)}
```
Comment (larkox) [inline webapp/channels/src/components/file_preview_modal/file_preview_modal_main_actions/file_preview_modal_main_actions.tsx:100]: Can you verify what is the effect of this and what would be the problem if we were to remove this line?

File: webapp/channels/src/components/with_tooltip/index.tsx [hunk 2]
```diff
@@ -28,6 +29,8 @@ const WithTooltip = ({
     placement,
     onShow,
     children,
+    onExit,
+    shouldUpdatePosition,
```
Comment (larkox) [inline webapp/channels/src/components/with_tooltip/index.tsx:33]: What does this do? What is the impact if we remove it?

**Review by larkox — COMMENTED** [review]
A couple of questions.

**Comment by M-ZubairAhmed** [comment]
@Aryakoste let us know if you need help in addressing the review comments

**Comment by Aryakoste** [comment]
If we take it from code perspective the title preview displays the main message of tooltip. So if we remove then we will not get any text as to 'Public link copied' or 'Get a public link'. When we copy the public link by clicking on tooltip then the message changes and when we exit the tooltip the message changes. So if we remove the onExit then this will not happen there will be only one message. The shouldUpdatePosition allows the tooltip to be repositioned whenever the content inside of tooltip gets changed in this case it is. @larkox @M-ZubairAhmed

**Comment by Aryakoste** [comment]
As the tooltip text needs to get changed from copied to get the public link.

**Comment by larkox** [comment]
@Aryakoste Thank you for the details! It makes sense.

@M-ZubairAhmed I don't love how this was originally implemented, but kind of makes sense. 0/5 if you want to give it another thought in case you want to refactor it so we don't have to pass so many new props to "WithTooltip". But these ones sound ok to add to me.

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
> @Aryakoste Thank you for the details! It makes sense.
> 
> @M-ZubairAhmed I don't love how this was originally implemented, but kind of makes sense. 0/5 if you want to give it another thought in case you want to refactor it so we don't have to pass so many new props to "WithTooltip". But these ones sound ok to add to me.

@larkox i compiled all these comments in the ticket when we are swapping out the internal implementation for take a look at.

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**